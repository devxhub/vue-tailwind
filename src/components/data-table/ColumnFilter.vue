<template>
  <div class="p-2 absolute z-1 bg-white shadow-md top-full right-0 mt-1">
    <div class="overflow-hidden flex flex-col font-normal" @click.stop="close">
      <button type="button" :class="{ active: props.column.condition === '' }" @click="select('')">
        {{ (props.columnFilterLang && props.columnFilterLang['no_filter']) ?? 'No filter' }}
      </button>
      <template v-if="props.column.type === 'string'">
        <button
          type="button"
          :class="{ active: props.column.condition === 'contain' }"
          @click="select('contain')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['contain']) ?? 'Contain' }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'not_contain' }"
          @click="select('not_contain')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['not_contain']) ?? 'Not contain' }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'equal' }"
          @click="select('equal')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['equal']) ?? 'Equal' }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'not_equal' }"
          @click="select('not_equal')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['not_equal']) ?? 'Not equal' }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'start_with' }"
          @click="select('start_with')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['start_with']) ?? 'Starts with' }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'end_with' }"
          @click="select('end_with')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['end_with']) ?? 'Ends with' }}
        </button>
      </template>
      <template v-else-if="props.column.type === 'number'">
        <button
          type="button"
          :class="{ active: props.column.condition === 'equal' }"
          @click="select('equal')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['equal']) ?? 'Equal' }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'not_equal' }"
          @click="select('not_equal')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['not_equal']) ?? 'Not equal' }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'greater_than' }"
          @click="select('greater_than')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['greater_than']) ?? 'Greater than' }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'greater_than_equal' }"
          @click="select('greater_than_equal')"
        >
          {{
            (props.columnFilterLang && props.columnFilterLang['greater_than_equal']) ??
            'Greater than or equal'
          }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'less_than' }"
          @click="select('less_than')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['less_than']) ?? 'Less than' }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'less_than_equal' }"
          @click="select('less_than_equal')"
        >
          {{
            (props.columnFilterLang && props.columnFilterLang['less_than_equal']) ??
            'Less than or equal'
          }}
        </button>
      </template>
      <template v-else-if="props.column.type === 'date'">
        <button
          type="button"
          :class="{ active: props.column.condition === 'equal' }"
          @click="select('equal')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['equal']) ?? 'Equal' }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'not_equal' }"
          @click="select('not_equal')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['not_equal']) ?? 'Not equal' }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'greater_than' }"
          @click="select('greater_than')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['greater_than']) ?? 'Greater than' }}
        </button>
        <button
          type="button"
          :class="{ active: props.column.condition === 'less_than' }"
          @click="select('less_than')"
        >
          {{ (props.columnFilterLang && props.columnFilterLang['less_than']) ?? 'Less than' }}
        </button>
      </template>

      <button
        type="button"
        :class="{ active: props.column.condition === 'is_null' }"
        @click="select('is_null')"
      >
        {{ (props.columnFilterLang && props.columnFilterLang['is_null']) ?? 'Is null' }}
      </button>
      <button
        type="button"
        :class="{ active: props.column.condition === 'is_not_null' }"
        @click="select('is_not_null')"
      >
        {{ (props.columnFilterLang && props.columnFilterLang['is_not_null']) ?? 'Not null' }}
      </button>
    </div>
  </div>
</template>
<script setup lang="ts">
import { onBeforeUnmount, onMounted } from 'vue'

const props = defineProps(['column', 'columnFilterLang'])

onBeforeUnmount(() => {
  document.removeEventListener('click', close)
})
onMounted(() => {
  document.addEventListener('click', close)
})
const emit = defineEmits(['close', 'filterChange'])

const close = () => {
  emit('close')
}

const select = (condition: any) => {
  // eslint-disable-next-line vue/no-mutating-props
  props.column.condition = condition
  if (condition === '') {
    // eslint-disable-next-line vue/no-mutating-props
    props.column.value = ''
  }

  emit('filterChange', props.column)
}
</script>

<style scoped>
button {
  width: 100%;
  text-align: left;
}
button:hover {
  background-color: rgb(226 232 240);
}
</style>
