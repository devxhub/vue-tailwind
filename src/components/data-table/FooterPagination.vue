<template>
  <div class="py-5" :class="{ 'pointer-events-none': currentLoader }">
    <div class="flex items-center flex-wrap flex-col sm:flex-row gap-4">
      <div class="flex items-center bh-pagination-info">
        <span class="mr-2">
          {{
            stringFormat(props.paginationInfo, filterRowCount ? offset : 0, limit, filterRowCount)
          }}
        </span>
        <select
          v-if="props.showPageSize"
          v-model="currentPageSize"
          @change="emit('changePageSize', currentPageSize)"
        >
          <option v-for="option in props.pageSizeOptions" :value="option" :key="option">
            {{ option }}
          </option>
        </select>
      </div>

      <div class="sm:ml-auto inline-flex items-center space-x-2">
        <button
          v-if="props.showFirstPage"
          type="button"
          class="bh-page-item first-page"
          :class="{ disabled: currentPage <= 1 }"
          @click="(currentPage = 1), emit('changePage', currentPage)"
        >
          <span v-if="props.firstArrow" v-html="props.firstArrow"> </span>
          <slot v-else name="first-arrow">
            <FirstArrow />
          </slot>
        </button>
        <button
          type="button"
          class="bh-page-item previous-page"
          :class="{ disabled: currentPage <= 1 }"
          @click="previousPage"
        >
          <span v-if="props.previousArrow" v-html="props.previousArrow"> </span>
          <PrevArrow  />
        </button>

        <template v-if="props.showNumbers">
          <button
            v-for="page in paging"
            :key="page"
            type="button"
            class="bh-page-item"
            :class="{
              disabled: currentPage === page,
              'bh-active': page === currentPage
            }"
            @click="movePage(page)"
          >
            {{ page }}
          </button>
        </template>

        <button
          type="button"
          class="bh-page-item next-page"
          :class="{ disabled: currentPage >= maxPage }"
          @click="nextPage"
        >
          <span v-if="props.nextArrow" v-html="props.nextArrow"> </span>
          <LastArrow v-else />
        </button>

        <button
          v-if="props.showLastPage"
          type="button"
          class="bh-page-item last-page"
          :class="{ disabled: currentPage >= maxPage }"
          @click="(currentPage = maxPage), emit('changePage', currentPage)"
        >
          <span v-if="props.lastArrow" v-html="props.lastArrow"> </span>
          <NextArrow v-else />
        </button>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { computed, ref, watch, defineAsyncComponent } from 'vue'
const FirstArrow = defineAsyncComponent(() => import('@/assets/icons/FirstArrow.vue'))
const PrevArrow = defineAsyncComponent(() => import('@/assets/icons/PrevArrow.vue'))
const LastArrow = defineAsyncComponent(() => import('@/assets/icons/LastArrow.vue'))
const NextArrow = defineAsyncComponent(() => import('@/assets/icons/NextArrow.vue'))

const props = withDefaults(
  defineProps<{
    currentLoader?: boolean
    totalRows?: number
    rows?: Array<any>
    offset?: number
    limit?: number
    page?: number
    paginationInfo?: string // default: "Showing {0} to {1} of {2} entries"
    showPageSize: boolean
    pageSizeOptions?: Array<number> // default: [10, 20, 30, 50, 100]
    showFirstPage?: boolean
    showLastPage?: boolean
    firstArrow?: string
    lastArrow?: string
    nextArrow?: string
    previousArrow?: string
    showNumbers?: boolean
    showNumbersCount?: number
    pageSize?: number // default: 10
    pagination?: boolean
    isServerMode?: boolean
  }>(),
  {
    currentLoader: false,
    totalRows: 0,
    rows: () => [],
    offset: 0,
    limit: 0,
    page: 1,
    paginationInfo: 'Showing {0} to {1} of {2} entries',
    showPageSize: true,
    pageSizeOptions: () => [10, 20, 30, 50, 100],
    showFirstPage: true,
    showLastPage: true,
    firstArrow: '',
    lastArrow: '',
    nextArrow: '',
    previousArrow: '',
    showNumbers: true,
    showNumbersCount: 5,
    pageSize: 10,
    pagination: true,
    isServerMode: false
  }
)

const emit: any = defineEmits(['changePageSize', 'changePage'])

const currentPage = ref(props?.page)
const currentPageSize = ref(props.pagination ? props?.pageSize : props?.rows?.length)
const filterRowCount = ref(props?.totalRows)

const stringFormat = (template: string, ...args: any[]) => {
  return template.replace(/{(\d+)}/g, (match, number) => {
    return typeof args[number] != 'undefined' ? args[number] : match
  })
}

// Maximum number of pages
const maxPage = computed(() => {
  const totalPages =
    <number>currentPageSize?.value < 1
      ? 1
      : Math.ceil(<number>filterRowCount.value / <number>currentPageSize?.value)
  return Math.max(totalPages || 0, 1)
})

// Paging array
const paging = computed(() => {
  let startPage: number, endPage: number
  let isMaxSized =
    typeof props.showNumbersCount !== 'undefined' && <number>props.showNumbersCount < maxPage.value
  // recompute if maxSize
  if (isMaxSized) {
    // Current page is displayed in the middle of the visible ones
    startPage = Math.max(currentPage.value - Math.floor(<number>props.showNumbersCount / 2), 1)
    endPage = startPage + <number>props.showNumbersCount - 1

    // Adjust if limit is exceeded
    if (endPage > maxPage.value) {
      endPage = maxPage.value
      startPage = endPage - <number>props.showNumbersCount + 1
    }
  } else {
    startPage = 1
    endPage = maxPage.value
  }

  const pages = Array.from(Array(endPage + 1 - startPage).keys()).map((i) => startPage + i)

  return pages
})

// page change
const movePage = (page: number) => {
  currentPage.value = page
  emit('changePage', page)
}

// next page
const nextPage = () => {
  if (currentPage.value >= maxPage.value) {
    return false
  }
  currentPage.value++
  emit('changePage', currentPage.value)
}

// previous page
const previousPage = () => {
  if (currentPage.value == 1) {
    return false
  }
  currentPage.value--
  emit('changePage', currentPage.value)
}
</script>
