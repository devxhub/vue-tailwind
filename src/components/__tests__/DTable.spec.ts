import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import ColumnHeader from '../data-table/ColumnHeader.vue'
import ColumnFilter from '../data-table/ColumnFilter.vue'
import FooterPagination from '../data-table/FooterPagination.vue'
import DTable from '../data-table/DTable.vue'

describe('ColumnFilter.vue', () => {
  // Mock data for testing
  const propsData = {
    column: {
      type: 'string',
      condition: 'contain',
      value: ''
    },
    columnFilterLang: {
      no_filter: 'No filter',
      contain: 'Contain',
      not_contain: 'Not contain',
      equal: 'Equal',
      not_equal: 'Not equal',
      start_with: 'Starts with',
      end_with: 'Ends with',
      greater_than: 'Greater than',
      greater_than_equal: 'Greater than or equal',
      less_than: 'Less than',
      less_than_equal: 'Less than or equal',
      is_null: 'Is null',
      is_not_null: 'Not null'
    }
  }

  const wrapper: any = mount(ColumnFilter, { props: propsData })

  // Test for rendering with correct props
  it('renders with correct props', async () => {
    // Check if the component renders with the correct props
    expect(wrapper.props()).toEqual(propsData)
  })

  // Test for selecting a filter condition
  it('emits filterChange event when a filter condition is selected', async () => {
    // You can simulate the user selecting a filter condition, e.g., 'contain'
    await wrapper.find('button.active').trigger('click')

    // Check if the filterChange event is emitted with the correct data
    expect(wrapper.emitted().filterChange).toBeTruthy()
    expect(wrapper.emitted().filterChange[0][0]).toEqual({
      type: 'string',
      condition: 'contain',
      value: ''
    })
  })
})

describe('ColumnHeader.vue', () => {
  // Mock data for testing
  const propsData = {
    all: {
      hasCheckbox: true,
      stickyHeader: true,
      stickyFirstColumn: false,
      columns: [
        { field: 'id', title: 'ID', hide: false, sort: true, width: '100px' },
        { field: 'name', title: 'Name', hide: false, sort: true, width: '150px' }
        // Add more sample columns as needed
      ]
      // Add other props as needed
    },
    currentSortColumn: 'id',
    currentSortDirection: 'asc',
    isOpenFilter: null,
    isFooter: false,
    checkAll: 0,
    columnFilterLang: 'en' // Replace with your language preference
  }

  const wrapper = mount(ColumnHeader, { props: propsData })

  // Test for rendering with correct props
  it('renders with correct props', async () => {
    // Check if the component renders with the correct props
    expect(wrapper.props()).toEqual(propsData)
  })

  // Test for selectAll event
  it('emits selectAll event when checkbox is clicked', async () => {
    await wrapper.find('#select').trigger('click')
    expect(wrapper.emitted().selectAll).toBeTruthy()
  })
})

describe('FooterPagination.vue', () => {
  it('renders correctly', () => {
    const props = {
      totalRows: 100,
      offset: 1,
      limit: 10,
      page: 1,
      showPageSize: true,
      pageSizeOptions: [10, 20, 30]
    }

    const wrapper = mount(FooterPagination, { props })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('displays the correct initial page information', () => {
    const props = {
      totalRows: 100,
      offset: 1,
      limit: 10,
      page: 1,
      showPageSize: true,
      pageSizeOptions: [10, 20, 30]
    }

    const wrapper = mount(FooterPagination, { props })

    expect(wrapper.find('.bh-pagination-info').text()).toContain('Showing 1 to 10 of 100 entries')
  })

  it('emits "changePage" event when a page button is clicked', async () => {
    const props = {
      totalRows: 100,
      offset: 1,
      limit: 10,
      page: 1,
      showPageSize: true,
      pageSizeOptions: [10, 20, 30]
    }

    const wrapper: any = mount(FooterPagination, { props })

    await wrapper.find('.bh-page-item:nth-child(1)').trigger('click')

    expect(wrapper.emitted().changePage).toBeTruthy()
    expect(wrapper.emitted().changePage[0][0]).toBe(1)
  })

  it('emits "changePage" event when "Next" button is clicked', async () => {
    const props = {
      totalRows: 100,
      offset: 1,
      limit: 10,
      page: 1,
      showPageSize: true,
      pageSizeOptions: [10, 20, 30]
    }

    const wrapper: any = mount(FooterPagination, { props })

    await wrapper.find('.bh-page-item.next-page').trigger('click')

    expect(wrapper.emitted().changePage).toBeTruthy()
    expect(wrapper.emitted().changePage[0][0]).toBe(2)
  })

  it('emits "changePage" event when "Previous" button is clicked', async () => {
    const props = {
      totalRows: 100,
      offset: 1,
      limit: 10,
      page: 2,
      showPageSize: true,
      pageSizeOptions: [10, 20, 30]
    }

    const wrapper: any = mount(FooterPagination, { props })

    await wrapper.find('.bh-page-item.previous-page').trigger('click')

    expect(wrapper.emitted().changePage).toBeTruthy()
    expect(wrapper.emitted().changePage[0][0]).toBe(1)
  })

  it('emits "changePage" event when "First" button is clicked', async () => {
    const props = {
      totalRows: 100,
      offset: 1,
      limit: 10,
      page: 2,
      showPageSize: true,
      pageSizeOptions: [10, 20, 30]
    }

    const wrapper: any = mount(FooterPagination, { props })

    await wrapper.find('.bh-page-item.first-page').trigger('click')

    expect(wrapper.emitted().changePage).toBeTruthy()
    expect(wrapper.emitted().changePage[0][0]).toBe(1)
  })

  it('emits "changePage" event when "Last" button is clicked', async () => {
    const props = {
      totalRows: 100,
      offset: 1,
      limit: 10,
      page: 1,
      showPageSize: true,
      pageSizeOptions: [10, 20, 30]
    }

    const wrapper: any = mount(FooterPagination, { props })

    await wrapper.find('.bh-page-item.last-page').trigger('click')

    expect(wrapper.emitted().changePage).toBeTruthy()
    expect(wrapper.emitted().changePage[0][0]).toBe(10)
  })
})

describe('DTable.vue', () => {
  it('renders with correct props and emits change event', async () => {
    // Mock data for testing
    const propsData = {
      loading: false,
      isServerMode: false,
      skin: 'bh-table-striped bh-table-hover',
      totalRows: 100,
      rows: [
        { id: 1, name: 'John Doe', age: 25 }
        // Add more sample rows as needed
      ],
      columns: [
        { field: 'id', label: 'ID' },
        { field: 'name', label: 'Name' },
        { field: 'age', label: 'Age' }
        // Add more sample columns as needed
      ],
      hasCheckbox: true,
      search: 'Sample Search',
      columnChooser: false,
      page: 1,
      pageSize: 10,
      pageSizeOptions: [10, 20, 30, 50, 100],
      showPageSize: true,
      rowClass: null,
      cellClass: null,
      sortable: true,
      sortColumn: 'id',
      sortDirection: 'asc',
      columnFilter: true,
      columnFilterLang: null,
      pagination: true,
      showNumbers: true,
      showNumbersCount: 5,
      showFirstPage: true,
      showLastPage: true,
      firstArrow: '',
      lastArrow: '',
      nextArrow: '',
      previousArrow: '',
      paginationInfo: 'Showing {0} to {1} of {2} entries',
      noDataContent: 'No data available',
      stickyHeader: false,
      height: '500px',
      stickyFirstColumn: false,
      cloneHeaderInFooter: false,
      selectRowOnClick: false
    }

    const wrapper = mount(DTable, { props: propsData })
    // Check if the component renders with the correct props
    expect(wrapper.props()).toEqual(propsData)
  })

  // Test for the "change" event
  it('emits the "change" event when triggered', async () => {
    const wrapper = mount(DTable)
    await wrapper.vm.$emit('change')
    expect(wrapper.emitted().change).toBeTruthy()
  })

  // Test for the "sortChange" event
  it('emits the "sortChange" event when triggered', async () => {
    const wrapper = mount(DTable)
    await wrapper.vm.$emit('sortChange')
    expect(wrapper.emitted().sortChange).toBeTruthy()
  })

  // Test for the "searchChange" event
  it('emits the "searchChange" event when triggered', async () => {
    const wrapper = mount(DTable)
    await wrapper.vm.$emit('searchChange')
    expect(wrapper.emitted().searchChange).toBeTruthy()
  })

  // Test for the "pageChange" event
  it('emits the "pageChange" event when triggered', async () => {
    const wrapper = mount(DTable)
    await wrapper.vm.$emit('pageChange')
    expect(wrapper.emitted().pageChange).toBeTruthy()
  })

  // Test for the "pageSizeChange" event
  it('emits the "pageSizeChange" event when triggered', async () => {
    const wrapper = mount(DTable)
    await wrapper.vm.$emit('pageSizeChange')
    expect(wrapper.emitted().pageSizeChange).toBeTruthy()
  })

  // Test for the "rowSelect" event
  it('emits the "rowSelect" event when triggered', async () => {
    const wrapper = mount(DTable)
    await wrapper.vm.$emit('rowSelect')
    expect(wrapper.emitted().rowSelect).toBeTruthy()
  })

  // Test for the "filterChange" event
  it('emits the "filterChange" event when triggered', async () => {
    const wrapper = mount(DTable)
    await wrapper.vm.$emit('filterChange')
    expect(wrapper.emitted().filterChange).toBeTruthy()
  })

  // Test for the "rowClick" event
  it('emits the "rowClick" event when triggered', async () => {
    const wrapper = mount(DTable)
    await wrapper.vm.$emit('rowClick')
    expect(wrapper.emitted().rowClick).toBeTruthy()
  })

  // Test for the "rowDBClick" event
  it('emits the "rowDBClick" event when triggered', async () => {
    const wrapper = mount(DTable)
    await wrapper.vm.$emit('rowDBClick')
    expect(wrapper.emitted().rowDBClick).toBeTruthy()
  })

  // Test for the "changePageSize" event
  it('emits the "changePageSize" event when triggered', async () => {
    const wrapper = mount(DTable)
    await wrapper.vm.$emit('changePageSize')
    expect(wrapper.emitted().changePageSize).toBeTruthy()
  })

  // Test for the "changePage" event
  it('emits the "changePage" event when triggered', async () => {
    const wrapper = mount(DTable)
    await wrapper.vm.$emit('changePage')
    expect(wrapper.emitted().changePage).toBeTruthy()
  })

  it('renders table with correct number of rows', async () => {
    const wrapper = mount(DTable, {
      props: {
        rows: [
          { id: 1, name: 'John Doe', age: 30 },
          { id: 2, name: 'Jane Doe', age: 25 }
        ],
        columns: [
          { field: 'id', type: 'number' },
          { field: 'name', type: 'string' },
          { field: 'age', type: 'number' }
        ]
      }
    })

    await wrapper.vm.$nextTick()
    expect(wrapper.findAll('tbody tr').length).toBe(2)
  })
})
