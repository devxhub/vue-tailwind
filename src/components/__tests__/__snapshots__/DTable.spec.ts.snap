// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FooterPagination.vue > renders correctly 1`] = `
"<div class=\\"py-5\\">
  <div class=\\"flex items-center flex-wrap flex-col sm:flex-row gap-4\\">
    <div class=\\"flex items-center\\"><span class=\\"mr-2\\">Showing 1 to 10 of 100 entries</span><select>
        <option value=\\"10\\">10</option>
        <option value=\\"20\\">20</option>
        <option value=\\"30\\">30</option>
      </select></div>
    <div class=\\"sm:ml-auto inline-flex items-center space-x-2\\"><button type=\\"button\\" class=\\"bh-page-item first-page disabled\\">
        <!---->
      </button><button type=\\"button\\" class=\\"bh-page-item previous-page disabled\\">
        <!--v-if-->
        <!---->
      </button><button type=\\"button\\" class=\\"bh-page-item disabled bh-active\\">1</button><button type=\\"button\\" class=\\"bh-page-item\\">2</button><button type=\\"button\\" class=\\"bh-page-item\\">3</button><button type=\\"button\\" class=\\"bh-page-item\\">4</button><button type=\\"button\\" class=\\"bh-page-item\\">5</button><button type=\\"button\\" class=\\"bh-page-item next-page\\">
        <!---->
      </button><button type=\\"button\\" class=\\"bh-page-item last-page\\">
        <!---->
      </button></div>
  </div>
</div>"
`;
